hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/LiteralEndToEndTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+
+import static com.hazelcast.sql.SqlColumnType.BIGINT;
+import static com.hazelcast.sql.SqlColumnType.BOOLEAN;
+import static com.hazelcast.sql.SqlColumnType.DECIMAL;
+import static com.hazelcast.sql.SqlColumnType.DOUBLE;
+import static com.hazelcast.sql.SqlColumnType.INTEGER;
+import static com.hazelcast.sql.SqlColumnType.NULL;
+import static com.hazelcast.sql.SqlColumnType.SMALLINT;
+import static com.hazelcast.sql.SqlColumnType.TINYINT;
+import static com.hazelcast.sql.SqlColumnType.VARCHAR;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class LiteralEndToEndTest extends ExpressionEndToEndTestBase {
+
+    @Test
+    public void testValid() {
+        assertRow("0", EXPR0, TINYINT, (byte) 0);
+        assertRow("-0", EXPR0, TINYINT, (byte) 0);
+        assertRow("000", EXPR0, TINYINT, (byte) 0);
+        assertRow("1", EXPR0, TINYINT, (byte) 1);
+        assertRow("-1", EXPR0, TINYINT, (byte) -1);
+        assertRow("-01", EXPR0, TINYINT, (byte) -1);
+        assertRow("001", EXPR0, TINYINT, (byte) 1);
+        assertRow("100", EXPR0, TINYINT, (byte) 100);
+        assertRow(Byte.toString(Byte.MAX_VALUE), EXPR0, TINYINT, Byte.MAX_VALUE);
```

Comment:
The test doesn't assert `MIN_VALUE`-s. For example, `Byte.MAX_VALUE` yields `TINYINT`, while `Byte.MIN_VALUE` yields `SMALLINT`. Is there a reason for this?

Reply:
fixed in 25f20bbc325de647d3ae73f0ca58f80598a21258